apps/governance/src/app/pages/Stake/StakeForm.tsx

```diff
@@ -138,6 +140,22 @@ export const StakeForm: FC<Props> = ({ className, isMigrating = false }) => {
     )
   }
 
+  useEffect(() => {
+    const fetchSignature = async () => {
+      const address = await signer.getAddress()
+      if (!address) return
+
+      const signature = await fetch(`https://api.mstable.org/signature/${address}`)
+
+      const body = await signature.json()
+      setStakeSignatures(prevSignatures => ({
+        ...prevSignatures,
+        [address]: body.signature,
+      }))
+    }
+    fetchSignature()
+  }, [signer, setStakeSignatures])
+
```

Comment:
Not sure that this is the best place to call + assign data, might be worth pulling it out into a provider. Maybe limit calls too using the `FetchState` format that we sometimes use

Reply:
@chrisjgf do you think I should inject this in the [AccountProvider](https://github.com/mstable/mStable-apps/blob/962ec44be263620b60c47beba3aaf86d632ee802/libs/base/src/lib/context/AccountProvider.tsx)?

Reply:
Think that sounds good 👍 Maybe see what @JamesLefrere suggests

Reply:
perfect, I will check with him

Reply:
I think that's a good idea 👍